lib/rom/array_dataset.rb

```diff
@@ -0,0 +1,12 @@
+require 'rom/enumerable_dataset'
+
+module ROM
+  class ArrayDataset < EnumerableDataset
+    forward(
+      :*, :+, :-, :compact, :compact!, :delete_if, :flatten, :flatten!, :keep_if,
+      :map!, :length, :pop, :reject, :reject!, :reverse, :reverse!, :sample,
+      :select!, :size, :shift, :shuffle, :shuffle!, :slice, :slice!, :sort!,
+      :sort_by!, :uniq, :uniq!, :unshift, :values_at
```

Comment:
@elskwid well, we will have to keep an eye on Enumerable and Array methods. we'll see how it goes. Auto-forwarding of all methods would require a more complicated implementation (mostly because of to_enum stuff, otherwise we'd break semantics of enumerable/array). I didn't have fun listing those methods manually trust me, but I think it's the right thing to do. Needs to be explicit otherwise you may get funny stuff going on in your code (for instance we do not want to always wrap response with a dataset object, something we totally want a plain array back)